### ISSUE TITLE
Raise an error if a non-existing field-name is specified together with a correct type

### ISSUE THREAD

**Issue by joelostblom** [vega/altair#2825]
One thing I couldn't address in #2824 is when a non-existing column name is used with a correct data type. Currently that leads to an empty chart being created:

```py
import altair as alt
from vega_datasets import data

alt.Chart(data.cars()).mark_point().encode(x='Horsepowers:Q')
```

![image](https://user-images.githubusercontent.com/4560057/211322923-ad4fe8aa-4ef2-4b99-8f6c-4970726179d4.png)

It is not always easy to spot that there is a typo here, although there is some intuition in that if there is no data plotted then maybe there is an issue with the data column name. I don't know of an easy way to fix this because I think it suffers from all the same difficulties as mentioned in https://github.com/altair-viz/altair/pull/2586 that it is complicated to determine which column names should be allowed in the spec. This might ideally be solved on the Vega-Lite side of things.

**Comment by mattijn** [comment]
This can't be solved on the Vega-Lite side either since by design it doesn't _see_ the data.

**Comment by dangotbanned** [comment]
I don't think this would be possible without breaking expressions/transform_calculate
- https://altair-viz.github.io/user_guide/transform/calculate.html
- https://altair-viz.github.io/user_guide/interactions/bindings_widgets.html#encoding-channel-binding
- https://altair-viz.github.io/gallery/calculate_residuals.html
- https://altair-viz.github.io/gallery/waterfall_chart.html

Please feel free to reopen if you see a way around it!
